glusterd2/commands/snapshot/snapshot-restore.go

```diff
@@ -256,25 +271,39 @@ func storeSnapRestore(c transaction.TxnCtx) error {
 
 	newVolinfo := createRestoreVolinfo(snapInfo, vol)
 
+	// Volfile of restored volume
+	err = volgen.VolumeVolfileToStore(&newVolinfo, newVolinfo.Name, "client")
+	if err != nil {
+		c.Logger().WithError(err).WithFields(log.Fields{
+			"template": "client",
+			"volfile":  newVolinfo.Name,
+		}).Error("failed to generate volfile and save to store")
+		return err
+	}
+
 	if err := volume.AddOrUpdateVolumeFunc(&newVolinfo); err != nil {
 		c.Logger().WithError(err).WithField(
-			"volume", newVolinfo.Name).Debug("failed to store volume info")
+			"volume", newVolinfo.Name).Error("failed to store volume info")
 		return err
 	}
 
 	if err := snapshot.DeleteSnapshot(snapInfo); err != nil {
 		c.Logger().WithError(err).WithField(
-			"snapshot", snapVol.Name).Debug("failed to delete snap from store")
+			"snapshot", snapVol.Name).Error("failed to delete snap from store")
+		return err
 	}
 	if err := volgen.DeleteVolfiles(snapInfo.SnapVolinfo.VolfileID); err != nil {
 		c.Logger().WithError(err).
 			WithField("snapshot", snapshot.GetStorePath(snapInfo)).
 			Warn("failed to delete volfiles of snapshot")
+		return err
 	}
 
-	if err := volgen.Generate(); err != nil {
-		c.Logger().WithError(err).WithField(
-			"volume", newVolinfo.Name).Debug("failed to generate volfiles")
+	if err := volgen.DeleteBricksVolfiles(snapInfo.SnapVolinfo.GetLocalBricks()); err != nil {
```

Comment:
Can we also have a code to regenrate the same during the undo process. (glusterd2/commands/snapshot/snapshot-restore.go:undoSnapStore) This is needed only if the snapshot volume was started.

Actually, deletion is also required only when the snapshot was started. It's good to add the check here as well (Not mandatory, since it returns null if the file is not found).

Reply:
Fixed now, please review